File: Source/prj/main/ProjectTarget.py

class ProjectTarget:
    target: str
    tag: str

    def __init__(self, target: str, tag: str=None):
        self.target = target
        self.tag = tag

    def ToPath(self):
        if self.tag is None or self.tag == "":
            return self.target
        else:
            return self.target + "-" + self.tag

    def ToName(self):
        if self.tag is None or self.tag == "":
            return self.target
        else:
            return self.target + "; " + self.tag

    def __eq__(self, other):
        return self.__dict__ == other.__dict__

File: UnityPlugin/Projeny/Main/Models/ProjectTarget.cs
using System.Globalization;

namespace Projeny
{
    public struct ProjectTarget
    {
        public string Target;
        public string Tag;

        public object ToArgument()
        {
            if (string.IsNullOrEmpty(Tag))
            {
                return Target.ToLower(CultureInfo.InvariantCulture);
            }

            return $"{Target.ToLower(CultureInfo.InvariantCulture)}-{Tag}";
        }
    }
}